test/solution_candidate_selection_test.go

```diff
@@ -1,21 +1,38 @@
 package test
 
 import (
+	"github.com/weihesdlegend/Vacation-planner/POI"
+	"github.com/weihesdlegend/Vacation-planner/matching"
 	"github.com/weihesdlegend/Vacation-planner/solution"
 	"testing"
 )
 
 // test if priority queue gives top solution candidates
 func TestSolutionCandidateSelection(t *testing.T) {
-	candidates := make([]solution.PlanningSolution, 0)
+	places := make([]matching.Place, 0)
 
 	for i := 0; i < 100; i++ {
-		candidates = append(candidates, solution.PlanningSolution{Score: float64(i)})
+		places = append(places, matching.Place{Place: &POI.Place{Rating: float32(i), UserRatingsTotal: 9}, Price: 1})
 	}
 
-	res := solution.FindBestPlanningSolutions(candidates, 0)
+	iterator := &solution.MultiDimIterator{}
 
-	expectation := []float64{95, 96, 97, 98, 99}
+	clusters := make([][]matching.Place, 1)
+	clusters[0] = places
+	err := iterator.Init([]POI.PlaceCategory{POI.PlaceCategoryEatery}, clusters)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	topSolutionsCount := int64(5)
+	res := solution.FindBestPlanningSolutions(clusters, topSolutionsCount, iterator)
+
+	if int64(len(res)) != topSolutionsCount {
+		t.Errorf("expected number of solutions equals %d, got %d", topSolutionsCount, len(res))
+		return
+	}
+
+	expectation := []float64{94, 95, 96, 97, 98}
```

Comment:
I am curious why the expected top 5 values become 94-98 in this updated test, instead of 95-99.

Reply:
Because the minimum score becomes 0 instead of 1 in the previous test setting.